Nightly CI: fully_shard() API breakage + nvidia-resiliency-ext version mismatch

## Nightly CI Failure — 2026-06-01

**Run:** https://github.com/NVIDIA-BioNeMo/bionemo-framework/actions/runs/26750345835

### Failure 1: `fully_shard()` API breakage (3 recipes)

The `megatron-core` `fully_shard()` function has removed keyword arguments that our recipes still pass:

- **recipes/vit** (`train.py`): passes `grad_reduce_in_fp32` and `preserve_fp32_weights`
- **recipes/esm2_native_te** (`train.py`): passes `check_for_nan_in_grad`
- **recipes/geneformer_native_te_mfsdp_fp8** (`train.py`): passes `check_for_nan_in_grad`

**Error:**
```
TypeError: fully_shard() got an unexpected keyword argument 'grad_reduce_in_fp32'
TypeError: fully_shard() got an unexpected keyword argument 'check_for_nan_in_grad'
```

**Root cause:** The installed `megatron-core` version in the CI container has updated the `fully_shard()` API, removing these parameters. The recipes need to be updated to match.

**Fix:** Remove or replace the deprecated kwargs in each recipe's `train.py`. Need to check the current `megatron-core` `fully_shard()` signature to determine if these features were removed or renamed.

### Failure 2: `nvidia-resiliency-ext` version mismatch (models/geneformer)

**Error:**
```
AssertionError: Minimum required nvidia-resiliency-ext package version is 0.6.0.
```

This triggers when importing `megatron.core` → `dist_checkpointing` → `strategies.torch` → `nvrx.py`. The installed `nvidia-resiliency-ext` in the CI container is below 0.6.0.

**Affected tests:**
- `test_geneformer_checkpoint_loss[Geneformer-V1-10M]`
- `test_geneformer_checkpoint_weight_compatibility[Geneformer-V1-10M]`
- `test_te_bert_layer_and_hf_bert_layer_similar_output_values_random_inputs`
- `test_geneformer_model_loss_validity`

**Fix:** Update the CI container image to include `nvidia-resiliency-ext >= 0.6.0`, or pin the megatron-core version.

### Summary

| Job | Failures | Root Cause |
|-----|----------|------------|
| unit-tests (models/geneformer) | 4 | nvidia-resiliency-ext < 0.6.0 |
| unit-tests (recipes/vit) | 3 | `fully_shard()` API change |
| unit-tests (recipes/esm2_native_te) | 8 | `fully_shard()` API change |
| unit-tests (recipes/geneformer_native_te_mfsdp_fp8) | 3 | `fully_shard()` API change |

**Note:** Workstation was offline when this was diagnosed — could not attempt automated fix via devcontainer.

--- a/bionemo-recipes/recipes/esm2_native_te/hydra_config/defaults.yaml
+++ b/bionemo-recipes/recipes/esm2_native_te/hydra_config/defaults.yaml
@@ -34,9 +34,6 @@ fully_shard_kwargs:
   zero_dp_strategy: "optim_grads_params"
   calculate_per_token_loss: false
   init_model_with_meta_device: ${use_meta_device}
-  check_for_nan_in_grad: true
-  grad_reduce_in_fp32: false
-  preserve_fp32_weights: true
   overlap_grad_reduce: true
   overlap_param_gather: true
   sync_model_each_microbatch: true

--- a/bionemo-recipes/recipes/geneformer_native_te_mfsdp_fp8/hydra_config/defaults.yaml
+++ b/bionemo-recipes/recipes/geneformer_native_te_mfsdp_fp8/hydra_config/defaults.yaml
@@ -31,9 +31,6 @@ training:
     zero_dp_strategy: "optim_grads_params"
     calculate_per_token_loss: false
     init_model_with_meta_device: true
-    check_for_nan_in_grad: true
-    grad_reduce_in_fp32: false
-    preserve_fp32_weights: true
     overlap_grad_reduce: true
     overlap_param_gather: true
     sync_model_each_microbatch: true

--- a/bionemo-recipes/recipes/vit/config/defaults.yaml
+++ b/bionemo-recipes/recipes/vit/config/defaults.yaml
@@ -55,8 +55,6 @@ fsdp:
     - torch.nn.LayerNorm
     - torch.nn.Linear
   outer_dp_sharding_strategy: "optim"
-  grad_reduce_in_fp32: false
-  preserve_fp32_weights: true
 
 training:
   steps: 10

--- a/bionemo-recipes/recipes/vit/config/vit_base_patch16_224.yaml
+++ b/bionemo-recipes/recipes/vit/config/vit_base_patch16_224.yaml
@@ -53,8 +53,6 @@ fsdp:
     - torch.nn.LayerNorm
     - torch.nn.Linear
   outer_dp_sharding_strategy: 1
-  grad_reduce_in_fp32: false
-  preserve_fp32_weights: true
 
 training:
   steps: 500

--- a/bionemo-recipes/recipes/vit/train.py
+++ b/bionemo-recipes/recipes/vit/train.py
@@ -95,10 +95,6 @@ def main(cfg) -> None:
             hybrid_fsdp_group=device_mesh["hsdp"].get_group(),
             # Load the model on device in shards to avoid OOM. Requires device("meta")-init for model.
             init_model_with_meta_device=cfg.fsdp.init_model_with_meta_device,
-            # Reduce gradients in FP32.
-            grad_reduce_in_fp32=cfg.fsdp.grad_reduce_in_fp32,
-            # Store distributed optimization state in FP32.
-            preserve_fp32_weights=cfg.fsdp.preserve_fp32_weights,
             # Sync model parameters and gradients each step. Allows for param and gradient mods after BWD
             # pass, but deactivates compute-communication overlap going into the subsequent training step.
             sync_model_each_microbatch=True,

